feat: 서클 별 도서 추천 기능 구현

nyj001012 opened the pull request:
### 개요
서클 별 도서 추천 기능 구현

nyj001012 committed:
feat: 42 API 필터 조건 추가

- cursus id가 21이 포함된 프로젝트만 가져오도록 수정

nyj001012 committed:
docs: 불필요한 파일 제거 및 파일 양식 변경

nyj001012 committed:
docs: 파일 삭제 취소 및 양식 변경

nyj001012 committed:
docs: 프로젝트 이름 대신 id를 저장

nyj001012 committed:
feat: Project 타입에 updated_at 추가

nyj001012 committed:
style: 오타 수정

nyj001012 committed:
feat(books): 사용자의 프로젝트 id 가져오는 서비스 추가

- cursus_info.json 파일을 읽어와, 사용자의 프로젝트가 속한 서클 번호 읽어오는 함수 추가
- 사용자의 프로젝트 정보를 읽고, 추천할 도서의 프로젝트 id를 반환하는 함수 추가

nyj001012 committed:
feat: 추천 도서에 사용할 프로젝트 id값 받아오는 서비스 함수 호출

nyj001012 committed:
fix: BookService로부터 서비스 함수 호출

nyj001012 committed:
docs(books): json 파일 변경

- books_with_cursus_info.json 이름 변경
- projects_info.json 파일 내 여러 리스트를 하나의 리스트로 합치기

nyj001012 committed:
feat(books): DTO 추가

- 프로젝트의 정보를 담은 ProjectInfo 타입 추가
- books_with_project_info.json을 나타내는 BooksWithProjectInfo 타입 추가
- 응답 값을 담을 BookListWithSubject 타입 추가

nyj001012 committed:
refactor(cursus): 타입 명시

nyj001012 committed:
feat(books): 로그인 한 사용자의 첫 도서 추천 화면 구현

- 컨트롤러 구현
- 서비스 구현
- 레포지토리 구현

nyj001012 committed:
refactor: API 분리

nyj001012 committed:
style: 오타 수정

nyj001012 committed:
feat(cursus): /cursus/recommend/books API 변경

- path params의 type 삭제
- path params의 circleOrSubject를 project로 변경
- 도서 목록 키 값 bookList에서 items로 변경
- meta에 사용자 맞춤 정보를 담은 "사용자 맞춤" 추가
- 불필요한 에러 응답 제거

scarf005 commented on backend/src/v1/cursus/cursus.controller.ts:
```diff
@@ -1,8 +1,45 @@
+
+let accessToken: string;
+
+export const recommandBook = async (
```
```suggestion
export const recommendBook = async (
```

nyj001012 replied:
수정했습니다, 피드백 감사합니다!

nyj001012 committed:
feat(cursus): /assets의 세 개의 파일을 읽어오는 readFile 추가

- /asset의 세 파일을 읽어와 내용을 저장하는 전역 변수 정의
- /assets의 세 파일을 읽어오는 함수 구현

nyj001012 committed:
refactor(cursus): DTO 타입 이름 변경

- RawProject -> UserProjectFrom42
- Project -> UserProject
- BookListWithSubject -> RecommendedBook

nyj001012 committed:
feat(cursus): meta의 0번째 인덱스 아이템 추가

- meta의 0번째 인덱스에 추천 도서를 나타내는 아이템 추가

nyj001012 committed:
refactor(cursus): list의 중복 제거 단순화

nyj001012 committed:
refactor(cursus): 지역변수 정리

nyj001012 committed:
refactor(cursus): 변수 및 함수명 변경

- bookIds -> bookInfoIds
- getRecommendedBookIds -> getRecommendedBookInfoIds

nyj001012 committed:
feat(cursus): 프로젝트 이름으로 추천 도서 검색하는 기능 추가

- 컨트롤러 추가
- 서비스 추가

nyj001012 committed:
refactor(cursus): 누락된 return 추가

nyj001012 committed:
fix(cursus): errorCode가 null로 보이는 것 수정

nyj001012 committed:
style(cursus): 주석 및 사용하지 않는 import 삭제

nyj001012 committed:
style(cursus): import 경로를 절대 경로로 변경

nyj001012 committed:
Merge branch 'develop' into 706-서클-별-도서-추천-기능-구현

nyj001012 committed:
fix: roleset.all에서 미인증 유저가 포함된 경우 처리

미인증 유저가 authvalidate를 호출하면 401에러 대신 user 정보를 모두 null로 채워서 반환

nyj001012 committed:
fix(cursus): 미인증 유저가 프로젝트 조회할 때 빈 배열이 반환되는 것 수정

nyj001012 committed:
feat(cursus): 셔플 추가

- 미인증 유저가 모든 프로젝트에 대해 추천 도서를 검색할 때, 결과를 셔플하여 보여줌

nyj001012 committed:
style: 불필요한 import 삭제

Jiwon-Woo reviewed (approved):
LGTM!
